spynnaker/pyNN/models/neural_projections/connectors/fixed_number_pre_connector.py

```diff
@@ -36,69 +35,168 @@ def get_delay_maximum(self):
+#                         permutation = numpy.arange(self._n_pre_neurons)
+#                         for i in range(0, self._n_pre_neurons - 1):
+#                             j = int(self._rng.next(
+#                                 n=1, distribution="uniform",
+#                                 parameters=self._rng_parameters))
+#                             (permutation[i], permutation[j]) = (
+#                                 permutation[j], permutation[i])
+#                         n += self._n_pre_neurons
+#                         if self._pre_neurons[m] is None:
+#                             self._pre_neurons[m] = permutation
+#                         else:
+#                             self._pre_neurons[m] = numpy.append(
+#                                 self._pre_neurons, permutation)
+#                     self._pre_neurons[m] = self._pre_neurons[m][:self._n_pre]
+
+                # Sort the neurons now that we have them
+                self._pre_neurons[m].sort()
+
+                # If verbose then output the list connected to this post-neuron
+                if self._verbose:
+                    print 'post-neuron ', m, ' connects to pre-neurons '
+                    print self._pre_neurons[m]
+                    numpy.savetxt(file_handle,
+                                  self._pre_neurons[m][None, :],
+                                  fmt=("%u,"*(self._n_pre-1)+"%u"))
+
         return self._pre_neurons
 
-    def _pre_neurons_in_slice(self, pre_vertex_slice):
+    def _pre_neurons_in_slice(self, pre_vertex_slice, n):
         pre_neurons = self._get_pre_neurons()
-        return pre_neurons[
-            (pre_neurons >= pre_vertex_slice.lo_atom) &
-            (pre_neurons <= pre_vertex_slice.hi_atom)]
 
-    def _is_connected(self, pre_vertex_slice):
-        return self._pre_neurons_in_slice(pre_vertex_slice).size > 0
+        # Take the nth array and get the bits from it we need
+        # for this pre-vertex slice
+        this_pre_neuron_array = pre_neurons[n]
+
+        return this_pre_neuron_array[
+            (this_pre_neuron_array >= pre_vertex_slice.lo_atom) &
+            (this_pre_neuron_array <= pre_vertex_slice.hi_atom)]
+
+    def _is_connected(self, pre_vertex_slice, n):
+        return self._pre_neurons_in_slice(pre_vertex_slice, n).size > 0
 
     def get_n_connections_from_pre_vertex_maximum(
             self, pre_slices, pre_slice_index, post_slices,
             post_slice_index, pre_vertex_slice, post_vertex_slice,
             min_delay=None, max_delay=None):
 
-        if not self._is_connected(pre_vertex_slice):
+        if not self._is_connected(pre_vertex_slice, 0):
             return 0
 
+        # the number of max connections is either n_pre
+        # or post_vertex_slice.n_atoms
+        n_connections = 0
+        if self._n_pre > post_vertex_slice.n_atoms:
+            n_connections = self._n_pre
+        else:
+            n_connections = post_vertex_slice.n_atoms
```

Comment:
I think it is always n_pre since it is n to all, so every post neuron is connect to from n_pre neurons, which might be repeated, but still this is the number of connections